tests/rbd_mirror/test_rbd_mirror_image_features.py

```diff
@@ -0,0 +1,102 @@
+import time
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """Verification of change in properties of image in Primary Pool
+    gets reflected on remote site upon modification.
+
+    Args:
+        **kw:
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+
+    Test case covered -
+    CEPH-9520 - When a both Pool and image based mirroring is established,
+    verify if change of any image features in primary site should reflect
+    on remote site upon modification.
+
+
+    Pre-requisites :
+    1. Two Clusters must be up and running to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files on each node.
+
+    Test Case Flow:
+    1. Create a pool on both clusters.
+    2. Configure mirroring (peer bootstrap) between two clusters.
+    3. Enable pool mode journal based mirroring on the pool respectively.
+    4. Verify if change of any image features in primary site gets mirrored in secondary site.
+    5. Enable image mode snapshot based mirroring on the pool respectively.
+    6. Verify if change of any image features in primary site gets mirrored in secondary site.
+    """
+
+    try:
+        log.info("Running RBD mirroring test case for image features reflection")
+        config = kw.get("config")
+        mirror1, mirror2 = [
+            rbdmirror.RbdMirror(cluster, config)
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+        poolname = mirror1.random_string() + "_tier_1_rbd_mirror_pool"
+        imagename = mirror1.random_string() + "_tier_1_rbd_mirror_image"
+        imagespec = poolname + "/" + imagename
+
+        image_feature = "object-map,fast-diff"
+
+        # enable journal based mirroring for pool mode
+        mirror1.initial_mirror_config(
+            mirror2,
+            poolname=poolname,
+            imagename=imagename,
+            imagesize=config.get("imagesize", "1G"),
+            io_total=config.get("io-total", "10"),
+            mode="pool",
+            mirrormode="journal",
+            **kw,
+        )
+
+        # Disable image features in primary cluster
+        mirror1.image_feature_disable(imagespec=imagespec, image_feature=image_feature)
+        # wait till image gets reflected to secondary
+        time.sleep(60)
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
```

Comment:
Were the features object-map and fast-diff enabled by default? So that when you verify that they have been disabled its the actual result of disabling and not default status showing up

Reply:
2.11. Enabling and disabling image features
The block device images, such as fast-diff, exclusive-lock, object-map, or deep-flatten, are enabled by default.

Reply:
Makes sense. Thanks for the clarity.